File: cinema-ticket-C#/Model/Cinema.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CinemaTicket.Model;

[Table("cinemas")]
public class Cinema
{
    public long Id { get; set; }

    [Required, MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [Range(1, int.MaxValue)]
    [Column("row_count")]
    public int Rows { get; set; }

    [Range(1, int.MaxValue)]
    public int SeatsPerRow { get; set; }

    public ICollection<Screening> Screenings { get; set; } = new List<Screening>();

    [Timestamp]
    public byte[] RowVersion { get; set; } = Array.Empty<byte>();
}

File: Backend-Java/src/main/java/com/cinema/ticketsystem/model/Cinema.java
package com.cinema.ticketsystem.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "cinemas")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Cinema {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @NotBlank
    @Size(max = 100)
    @Column(nullable = false)
    private String name;
    
    @Min(1)
    @Column(nullable = false)
    private Integer rows;
    
    @Min(1)
    @Column(nullable = false)
    private Integer seatsPerRow;
    
    @OneToMany(mappedBy = "cinema", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Screening> screenings = new ArrayList<>();
}
